### ISSUE TITLE
App crash if location result contains NaN

### ISSUE THREAD

**Issue by arq-lukeshan** [transistorsoft/flutter_background_geolocation#632]
## Your Environment
* Plugin version: 4.2.1
* Platform: Android
* OS version: 9
* Device manufacturer / model: Huawei Mate 20X
* Flutter info (`flutter doctor`):[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.2 20G95 darwin-x64, locale en-AU)
    • Flutter version 2.2.3 at /Users/luke/flutter
    • Framework revision f4abaa0735 (9 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4
* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
Config(
          preventSuspend: true,
          stationaryRadius: 25,
          heartbeatInterval: 60,
          desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
          distanceFilter: 10.0,
          stopOnTerminate: false,
          startOnBoot: true,
          enableHeadless: true,
          debug: false,
          logLevel: bg.Config.LOG_LEVEL_VERBOSE,
          reset: true)
```

## Steps to Reproduce

1. Install the apk on Huawei Mate 20X
2. Open the app
3. App crash
4.

## Debug logs

<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
═════════════════════════════════════════════
    ║ TrackingService: LocationResult
    ╠═════════════════════════════════════════════
    ╟─ 📍  Location[fused -28******,153****** hAcc=20 et=+3d4h16m8s486ms alt=62.0 vAcc=NaN sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}], age: 752831ms, time: 1630456531772
2021-09-01 10:48:04.708 16499-16544/xx.xxx.xxx.fwo E/TSLocationManager: [c.t.l.location.TSLocation buildJSONObject] 
      ‼️  JSON Errororg.json.JSONException: Forbidden numeric value: NaN
        at org.json.JSON.checkDouble(JSON.java:25)
        at org.json.JSONObject.put(JSONObject.java:270)
        at com.transistorsoft.locationmanager.location.TSLocation.buildJSONObject(Unknown Source:262)
        at com.transistorsoft.locationmanager.location.TSLocation.toJson(Unknown Source:4)
        at com.transistorsoft.locationmanager.location.TSLocation.renderJson(Unknown Source:39)
        at com.transistorsoft.locationmanager.data.sqlite.b.a(Unknown Source:24)
        at com.transistorsoft.locationmanager.data.sqlite.b.persist(Unknown Source:8)
        at com.transistorsoft.locationmanager.location.TSLocationManager$e.b(Unknown Source:52)
        at com.transistorsoft.locationmanager.location.TSLocationManager$e.run(Unknown Source:74)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:929)

```

</details>

Testing in Android emulator and it is working fine but on a Huawei Mate 20X it crash. Would be very appreciated if you can fix this issue.

**Comment by christocracy** [comment]
The issue template is not optional, it is required.  Edit your issue to provide the following information:

*****************************************************************
*
* WARNING:  DO NOT IGNORE THIS TEMPLATE
*
*****************************************************************



## Your Environment
* Plugin version:
* Platform: iOS or Android
* OS version:
* Device manufacturer / model:
* Flutter info (`flutter doctor`):
* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
PASTE_YOUR_CODE_HERE
```

## Expected Behavior


## Actual Behavior


## Steps to Reproduce

1.
2.
3.
4.

## Context


## Debug logs

<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
PASTE_YOUR_LOGS_HERE
```

</details>

**Comment by arq-lukeshan** [comment]
Issue updated. Thanks.

**Comment by christocracy** [comment]
> vAcc=NaN

```

JSONException: Forbidden numeric value: NaN
```

Huawei are **the worst** devices on earth.

**Comment by arq-lukeshan** [comment]
Would you be able to check the "NaN" value and replace it with some default value before saving it? This vertical accuracy is not important for geo fencing I believe.

**Comment by christocracy** [comment]
Released `4.2.2` with `NaN` check.

**Comment by arq-lukeshan** [comment]
Thank you! @christocracy
